### PR TITLE
AEDB-35 send provider interest email

### PR DESCRIPTION
**Pull request by scottywakefield**

### CODE REVIEW

File: src/SFA.DAS.EmployerDemand/Domain/Models/ProviderIsInterestedEmail.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            {
+                {"AEDEmployerName", recipientName },
+                {"AEDApprenticeshipTrainingCourse", $"{standardName} (level {standardLevel})" },
+                {"AEDApprenticeshipLocation", location },
+                {"AEDNumberOfApprentices", numberOfApprentices > 0 ? numberOfApprentices.ToString() : "Not sure" },
+                {"AEDProviderName", providerName },
+                {"AEDProviderEmail", providerEmail },
+                {"AEDProviderTelephone", providerPhone },
+                {"AEDProviderWebsite", !string.IsNullOrEmpty(providerWebsite) ? providerWebsite : "-" },
+                {"FatURL", providerOffersThisCourse ? $"https://findapprenticeshiptraining.apprenticeships.education.gov.uk/courses/{standardId}/providers/{ukprn}" : "---" },
```
Comment (dashton82) [inline src/SFA.DAS.EmployerDemand/Domain/Models/ProviderIsInterestedEmail.cs:37]: Id be tempted to pass this whole thing from the web - we have the URL for it and means its coming out of config as well then 💥

File: src/SFA.DAS.EmployerDemand/Application/ProviderInterest/Commands/CreateProviderInterests/CreateProviderInterestsCommandHandler.cs [hunk 2]
```diff
@@ -12,18 +15,46 @@ namespace SFA.DAS.EmployerDemand.Application.ProviderInterest.Commands.CreatePro
...
             _apiClient = apiClient;
+            _notificationService = notificationService;
         }
 
         public async Task<Guid> Handle(CreateProviderInterestsCommand request, CancellationToken cancellationToken)
         {
             var result = await _apiClient.PostWithResponseCode<PostCreateProviderInterestsResponse>(
                 new PostCreateProviderInterestsRequest(request));
 
+            if (result.StatusCode == System.Net.HttpStatusCode.Created)
```
Comment (dashton82) [inline src/SFA.DAS.EmployerDemand/Application/ProviderInterest/Commands/CreateProviderInterests/CreateProviderInterestsCommandHandler.cs:33]: Is the change done in the employer demand API to differentiate between created and accepted?

File: src/SFA.DAS.EmployerDemand.UnitTests/Domain/Models/WhenConstructingProviderIsInterestedEmail.cs [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+            string providerEmail,
+            string providerPhone,
+            string providerWebsite)
+        {
+            var expectedTokens = new Dictionary<string, string>
+            {
+                {"AEDEmployerName", recipientName },
+                {"AEDApprenticeshipTrainingCourse", $"{standardName} (level {standardLevel})" },
+                {"AEDApprenticeshipLocation", location },
+                {"AEDNumberOfApprentices", numberOfApprentices.ToString() },
```
Comment (dashton82) [inline src/SFA.DAS.EmployerDemand.UnitTests/Domain/Models/WhenConstructingProviderIsInterestedEmail.cs:31]: One to check - but is there the similar logic for the number of apprentices? as in if zero then dont show it?

Comment (scottywakefield) [reply src/SFA.DAS.EmployerDemand.UnitTests/Domain/Models/WhenConstructingProviderIsInterestedEmail.cs:31]: Doh, definitely meant to do that.

**Review by PeterDavidWilliams — DISMISSED** [review]

**Review by PeterDavidWilliams — APPROVED** [review]

**Pull request merged**